js/modules/k6/http/http_request_test.go

```diff
@@ -847,4 +847,27 @@ func TestRequestAndBatch(t *testing.T) {
 			assertRequestMetricsEmitted(t, state.Samples, "PUT", "https://httpbin.org/put", "", 200, "")
 		})
 	})
+
+	t.Run("HTTPRequest", func(t *testing.T) {
+		_, err := common.RunString(rt, `
+			let reqUrl = "https://httpbin.org/cookies"
+			let res = http.get(reqUrl);
+			let jar = new http.CookieJar();
+
+			jar.set("https://httpbin.org/cookies", "key", "value");
+			res = http.request("GET", "https://httpbin.org/cookies", null, { cookies: { key2: "value2" }, jar: jar });
+
+			if (res.json().cookies.key != "value") { throw new Error("wrong cookie value: " + res.json().cookies.key); }
+
+			if (res.status != 200) { throw new Error("wrong status: " + res.status); }
+			if (res.request["method"] !== "GET") { throw new Error("http request method was not \"GET\": " + JSON.stringify(res.request)) }
+			if (res.request["body"] != null) { throw new Error("http request body was not null: " + JSON.stringify(res.request)) }
```

Comment:
Missing test case for when body should be set. When testing this myself it seems that `request.body` does not work:
```
panic: reflect.Value.Addr of unaddressable value [recovered]--------] 0 / 1
        panic: Panic at 86: reflect.Value.Addr of unaddressable value [recovered]
        panic: Panic at 86: reflect.Value.Addr of unaddressable value

goroutine 37 [running]:
github.com/loadimpact/k6/vendor/github.com/dop251/goja.AssertFunction.func1.1(0xc422d2bd00)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/runtime.go:1407 +0xa2
panic(0xb53720, 0xc4201360d0)
        /home/markl/.gvm/gos/go1.9.2/src/runtime/panic.go:491 +0x283
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*vm).try.func1(0xc42203b450, 0x0, 0xc422d2bc08, 0x0, 0x0, 0x0, 0xc422d2bc60)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/vm.go:364 +0x4b4
panic(0xb2b620, 0xcdc1d0)
        /home/markl/.gvm/gos/go1.9.2/src/runtime/panic.go:491 +0x283
reflect.Value.Addr(0xb70f00, 0xc420609a30, 0x94, 0xc423036280, 0x1, 0x1)
        /home/markl/.gvm/gos/go1.9.2/src/reflect/value.go:239 +0x82
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*objectGoReflect)._getField(0xc42226e510, 0xc423036268, 0x6, 0x411eb7, 0xc4201360a0, 0x10)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/object_goreflect.go:93 +0xf8
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*objectGoReflect)._get(0xc42226e510, 0xc423036268, 0x6, 0x10, 0xc4201360a0)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/object_goreflect.go:111 +0xfc
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*objectGoReflect).getStr(0xc42226e510, 0xc423036268, 0x6, 0xc423055e20, 0x20)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/object_goreflect.go:124 +0x43
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*objectGoReflect).get(0xc42226e510, 0x1131f00, 0xc420136030, 0x18, 0xc423055e20)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/object_goreflect.go:86 +0x3f
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*_builtinJSON_stringifyContext).str(0xc421b8c820, 0x1131f00, 0xc420136030, 0xc4230551c0, 0x0)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/builtin_json.go:283 +0x6e
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*_builtinJSON_stringifyContext).jo(0xc421b8c820, 0xc4230551c0)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/builtin_json.go:458 +0x1f1
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*_builtinJSON_stringifyContext).str(0xc421b8c820, 0x1131f00, 0xc4203b9d20, 0xc423054900, 0x0)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/builtin_json.go:370 +0x3be
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*_builtinJSON_stringifyContext).jo(0xc421b8c820, 0xc423054900)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/builtin_json.go:458 +0x1f1
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*_builtinJSON_stringifyContext).str(0xc421b8c820, 0x1131f00, 0x12bfc00, 0xc4230556c0, 0x12bfc00)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/builtin_json.go:370 +0x3be
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*_builtinJSON_stringifyContext).do(0xc421b8c820, 0x1131d20, 0xc423054900, 0xc423036334)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/builtin_json.go:279 +0x1de
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*Runtime).builtinJSON_stringify(0xc422870480, 0x1131d20, 0xc42187ab80, 0xc423121e60, 0x3, 0x3, 0xc423055640, 0xc420c18500)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/builtin_json.go:270 +0x7d4
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*Runtime).(github.com/loadimpact/k6/vendor/github.com/dop251/goja.builtinJSON_stringify)-fm(0x1131d20, 0xc42187ab80, 0xc423121e60, 0x3, 0x3, 0x3, 0x4126f0)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/builtin_json.go:517 +0x51
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*Runtime).functionproto_apply(0xc422870480, 0x1131d20, 0xc42187abe0, 0xc421c9d450, 0x2, 0x39, 0xc423036310, 0xc422d2b8c0)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/builtin_function.go:59 +0x143
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*Runtime).(github.com/loadimpact/k6/vendor/github.com/dop251/goja.functionproto_apply)-fm(0x1131d20, 0xc42187abe0, 0xc421c9d450, 0x2, 0x39, 0x5, 0x1)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/builtin_function.go:159 +0x51
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*vm)._nativeCall(0xc42203b450, 0xc423004370, 0x2)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/vm.go:1826 +0x2ec
github.com/loadimpact/k6/vendor/github.com/dop251/goja.call.exec(0x2, 0xc42203b450)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/vm.go:1810 +0x6b3
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*call).exec(0xc42283c0ec, 0xc42203b450)
        <autogenerated>:1 +0x44
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*vm).run(0xc42203b450)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/vm.go:288 +0x51
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*funcObject).Call(0xc420ca1500, 0x1132360, 0x12e2100, 0x0, 0x0, 0x0, 0x456c17, 0xc42002c600)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/func.go:130 +0x3c8
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*funcObject).Call-fm(0x1132360, 0x12e2100, 0x0, 0x0, 0x0, 0x42a5c9, 0xc4212c0030)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/builtin_function.go:120 +0x51
github.com/loadimpact/k6/vendor/github.com/dop251/goja.AssertFunction.func1.2()
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/runtime.go:1412 +0xb6
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*vm).try(0xc42203b450, 0xc420049c78, 0x0)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/vm.go:370 +0x119
github.com/loadimpact/k6/vendor/github.com/dop251/goja.AssertFunction.func1(0x1132360, 0x12e2100, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/runtime.go:1411 +0x12e
github.com/loadimpact/k6/js.(*VU).RunOnce(0xc4216ac930, 0x112d240, 0xc4216d00c0, 0x0, 0xc4222d4bb0, 0xc420b979f0, 0xc420b95bb0, 0xc420b979e0)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/js/runner.go:260 +0x4c1
github.com/loadimpact/k6/core/local.(*vuHandle).run(0xc421517180, 0xc42007aaa0, 0xc42310e1e0, 0xc42310e180)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/core/local/local.go:64 +0x1cc
github.com/loadimpact/k6/core/local.(*Executor).scale.func1(0xc421517180, 0xc423177c20, 0xc42310e1e0, 0xc42310e180)
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/core/local/local.go:315 +0x4d
created by github.com/loadimpact/k6/core/local.(*Executor).scale
        /home/markl/.gvm/pkgsets/go1.9.2/global/src/github.com/loadimpact/k6/core/local/local.go:314 +0x35f
```

Reply:
oof, okay I'll look into this!